### DISCUSSION TITLE
Help me to understand correlation between phase A and B to CT Clamp

### DISCUSSION THREAD

**Discussion by catatung** [emporia-vue-local/esphome#233] [General]
Hi,

I've got an installation of the vue2 where phase A is configured to all even numbered ports and phase B is configured to all odd ports.

I'm seeing that, for example, a particular CT Clamp will only give a reading in HA if it is plugged into an even numbered port.  This suggests it is physically wired to Phase A.  Ok, I understand this completely.

What I don't understand though, is that why is the reading 0 when plugged into a port configured with the wrong phase?  It's my understanding that these CT clamps measure current inside the wire they are clamped over and the MCU multiplies that value with the voltage for the phase that port is configured to.  In a situation (such as mine) where both phases are connected and the HA interface has a reading for them (For example Phase A might read 119v while phase B might read 125v) I would think a misconfiguration would result in a slightly inaccurate measurement as a result of using the incorrect value between the two to calculate the power.  However, I am confused to see that using the wrong phase is resulting in a 0 value.

To summarize, it appears that if I have the phase incorrectly configured on a CT Clamp port number, I get a 0 reading instead of just a slightly inaccurate reading.  Both phases have a value and are measuring mains voltage.  Can anyone clarify why this is?

Thanks for any help
Randy

**Comment by flaviut** [comment]
This is caused by the filters field on each clamp, in particular `*pos`. The goal here is to avoid subtle errors by making them dramatic errors so folks fix their configs.

You can edit the filters field however you'd like though!

**Reply by catatung** [reply]
Hi Flaviut,

This make sense to me, however I'm curious how the vue2 ESP32 is able to identify whether or not the CT clamp is on a wire with the correct phase.  Does it correlate when the AC waveform is at v+ (120v or so) and expects the current reading for that moment to show load, and if this is not in sync then it ignores the reading and sets the value to 0?  I presume the incorrect phase would be 180 degrees out (inverse/opposite) of the correct phase.

**Reply by flaviut** [reply]
These are defined in the `.defaultfilters` section of the config. They are far less intelligent than you're expecting, and are implemented in just a single line of C++ code:

```
  - &pos
    # filter out any values below 0.
    lambda: 'return max(x, 0.0f);'
```

> I presume the incorrect phase would be 180 degrees out (inverse/opposite) of the correct phase.

Depends on where you live, and what kind of building you're in! But yes, this is correct if you live in a detached home in North America.

**Reply by catatung** [reply]
Thanks for taking the time to explain this, it makes complete sense.  It's nice to understand how and what is going on behind the scenes when troubleshooting a setup!  Or even if things are working correctly, it's nice to know what's going on behind it all.

Thanks again
